x-pack/platform/plugins/shared/triggers_actions_ui/public/application/sections/rule_details/components/rule_definition.tsx

```diff
@@ -112,13 +118,23 @@ export const RuleDefinition: React.FunctionComponent<RuleDefinitionProps> = memo
     }, [rule, ruleTypeRegistry]);
 
     const onEditRuleClick = () => {
-      navigateToApp('management', {
-        path: `insightsAndAlerting/triggersActions/${getEditRuleRoute(rule.id)}`,
-        state: {
-          returnApp: 'management',
-          returnPath: `insightsAndAlerting/triggersActions/${getRuleDetailsRoute(rule.id)}`,
-        },
-      });
+      if (ruleApp && ruleApp === 'observability') {
```

Comment:
For platform components, it is better not to have custom logic specific to a solution if possible. What about instead of passing a `ruleApp`, you pass a `navigateToEditRuleForm() => void` or similar, where the component can trigger. If the function is `undefined`, the component will fall back to the code as it was before.

Reply:
@cnasikas thank you for the suggestion. Updated in d5abf91e877cd235239824bb048bab1ffff92b1e